matrices.tex

```diff
@@ -25,6 +25,8 @@
 {x - sqrt(y*(x + y)): 1, x + sqrt(y*(x + y)): 1}
 \end{verbatim}
 
+%% TODO a Python list is not a dense representation...
+%% Is this supposed to say flat? - scopatz
```

Comment:
I don't think we are using different definitions when it comes to the matrices themselves. We both agree that a dense matrix stores all elements.  That said, I agree that we should be as specific as possible.  The LIL that you cited didn't mention a particular language. I'll add that Python is the weird case here where lists are not densely or contiguously stored.  

The polynomial usage is fine because it doesn't tie the densely causally to a Python data structure, which can be said to be non-dense. 

I think the matrix case could be a place where a reviewer would find this objectionable. It is better to be non-ambiguous and prevent such objections.